### ISSUE TITLE
Fix false positives for multi-line url() in max-line-length

### ISSUE THREAD

**Issue by AlexWayfer** [stylelint/stylelint#4059]
> Clearly describe the bug

There are false positives for max line length when there are spaces inside `url()` content.

> Which rule, if any, is the bug related to?

`max-line-length`

> What CSS is needed to reproduce the bug?

```css
div {
  background-image: url(
    "data:image/svg+xml;utf8,<svg fill='black' height='24' viewBox='0 0 24 24' width='24' xmlns='http://www.w3.org/2000/svg'><path d='M7 9l5 5 5-5z'/></svg>"
  );
}
```

> What tests are not passing?

```js
{
  code:
    "a {\n  background: url(\n    'something something  something'\n  );\n}"
}
```

> What stylelint configuration is needed to reproduce the bug?

```yaml
rules:
  max-line-length:
    - 20
```

> Which version of stylelint are you using?

`9.10.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "**/*.css"`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors did you get)?

The following warnings were flagged: 

```shell
70:102  ✖  Expected line length to be no more than 80 characters                                               max-line-length
```

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

Issue caused not because of space in url, but because `url("")` is multi-line.

This is not caused violations:

```css
div {
	background-image:
	url("data:image/svg+xml;utf8,<svg fill='black' height='24' viewBox='0 0 24 24' width='24' xmlns='http://www.w3.org/2000/svg'><path d='M7 9l5 5 5-5z'/></svg>");
}
div {
	background-image: url("data:image/svg+xml;utf8,<svg fill='black' height='24' viewBox='0 0 24 24' width='24' xmlns='http://www.w3.org/2000/svg'><path d='M7 9l5 5 5-5z'/></svg>");
}
```

This are causing false negative violations:

```css
div {
	background-image: url(
		"data:image/svg+xml;utf8,<svg fill='black' height='24' viewBox='0 0 24 24' width='24' xmlns='http://www.w3.org/2000/svg'><path d='M7 9l5 5 5-5z'/></svg>");
}
div {
	background-image: url("data:image/svg+xml;utf8,<svg fill='black' height='24' viewBox='0 0 24 24' width='24' xmlns='http://www.w3.org/2000/svg'><path d='M7 9l5 5 5-5z'/></svg>"
	);
}
```

As the issue has been labelled "help wanted", please consider [contributing a fix](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#fixing-a-bug-in-an-existing-rule).

**Comment by vankop** [comment]
Could I take this issue?

**Comment by hudochenkov** [comment]
@vankop Yes, please!

**Comment by vankop** [comment]
@hudochenkov As I understand #4057 has same problem, but with other rule?

**Comment by hudochenkov** [comment]
They might be related, because both have multi-line URLs.

**Comment by vankop** [comment]
@hudochenkov looks like it is postcss problem or problem how we use it 😁since it skip AST node if there is some not url encoded symbols (not all escaping is required)

Small repo with reproduction: https://github.com/vankop/stylelint-url-bug/tree/master

**Comment by hudochenkov** [comment]
PostCSS parses it. Check AST explorer https://astexplorer.net/#/gist/0914863cec899d3aa0b8063916d35c1e/c560f4a108e5b3c7803b88dfc4a293be3187cf5d. Issue in this RegExp:

https://github.com/stylelint/stylelint/blob/1178f2d2d4c7d83d12ad4ec55b686a7bf7e28995/lib/rules/max-line-length/index.js#L85

It doesn't match new line symbols.

**Comment by vankop** [comment]
we have test cases on it, see
https://github.com/stylelint/stylelint/blob/1178f2d2d4c7d83d12ad4ec55b686a7bf7e28995/lib/rules/max-line-length/__tests__/index.js#L34
code above checks new line symbols, see https://github.com/stylelint/stylelint/blob/1178f2d2d4c7d83d12ad4ec55b686a7bf7e28995/lib/rules/max-line-length/index.js#L66

I found out problem in debugging tests, I tried astexplorer and it works =).

**Comment by vankop** [comment]
@hudochenkov Steps to reproduce bug:
1. __lib/rules/max-line-length/__tests__/index.js__
```
"use strict";

const rule = require("..");
const { messages, ruleName } = rule;
const testDataUrl = "data:image/svg+xml;utf8,<svg fill='black' height='24' viewBox='0 0 24 24' width='24' xmlns='http://www.w3.org/2000/svg'><path d='M7 9l5 5 5-5z'/></svg>";

testRule(rule, {
  ruleName,
  config: [20],

  accept: [
    {
      code: `div {
        background-image: url("${testDataUrl}");
      }`
    },
    {
      code: `div {
        background-image: url("${encodeURI(testDataUrl)}");
      }`
    }
]
});
```
2. run ` node --inspect-brk node_modules/jest/bin/jest.js  --runInBand lib/rules/max-line-length/__tests__/index.js`
3. In debugger add breakpoint/`debugger;` here
https://github.com/stylelint/stylelint/blob/master/lib/getPostcssResult.js#L134

In first run `nodes: []`, in second run `nodes: [Node]`

**Comment by vankop** [comment]
> PostCSS parses it. Check AST explorer https://astexplorer.net/#/gist/0914863cec899d3aa0b8063916d35c1e/c560f4a108e5b3c7803b88dfc4a293be3187cf5d. Issue in this RegExp:
> 
> https://github.com/stylelint/stylelint/blob/1178f2d2d4c7d83d12ad4ec55b686a7bf7e28995/lib/rules/max-line-length/index.js#L85
> 
> It doesn't match new line symbols.

You are totally right, there is an issue with RegExp too. Current tests with newline in url works because of https://github.com/stylelint/stylelint/blob/master/lib/rules/max-line-length/index.js#L150
